drivers/power/lt8491/iio_lt8491.c

```diff
@@ -0,0 +1,361 @@
+		return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+	case LT8491_VIN:
+		ret = lt8491_read_vin(dev->dev, &uval);
+		if (ret)
+			return ret;
+
+		val = uval;
+
+		return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+	case LT8491_CHARGING_STAGE:
+		ret = lt8491_read_charging_stage(dev->dev, &val);
+		if (ret)
+			return ret;
+
+		return iio_format_value(buf, len, IIO_VAL_CHAR, 1, &val);
+	case LT8491_CHARGING_STATUS:
+		ret = lt8491_read_charging_status(dev->dev, &uval);
+		if (ret)
+			return ret;
+
+		val = uval;
+
+		return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+	case LT8491_ENABLE:
+		ret = lt8491_read_enable(dev->dev, &uval);
+		if (ret)
+			return ret;
+
+		val = uval;
+
+		return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+	case LT8491_SCRATCH:
+		ret = lt8491_read_scratch(dev->dev, &uval);
+		if (ret)
+			return ret;
+
+		val = uval;
+
+		return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+	case LT8491_SERIAL_ID:
+		ret = lt8491_read_serial_id(dev->dev, uvals);
+		if (ret)
+			return ret;
+
+		return iio_format_value(buf, len, IIO_VAL_INT_MULTIPLE,
+					NO_OS_ARRAY_SIZE(uvals),
+					(int32_t *)uvals);
+	default:
+		return -EOPNOTSUPP;
+	}
+}
+
+static int lt8491_ch_attr_store(void *ddev, char *buf,
+				uint32_t len,
+				const struct iio_ch_info *channel,
+				intptr_t priv)
+{
+	struct lt8491_iio_device *dev = ddev;
+	int32_t val;
+	int ret;
+
+	switch (priv) {
+	case LT8491_ENABLE:
+		ret = iio_parse_value(buf, IIO_VAL_INT, &val, NULL);
+		if (ret)
+			return ret;
+
+		if (val >> 1)
+			return -EINVAL;
+
+		ret = lt8491_write_enable(dev->dev, val);
+		if (ret)
+			return ret;
+
+		return len;
+	case LT8491_RESTART:
+		ret = iio_parse_value(buf, IIO_VAL_INT, &val, NULL);
+		if (ret)
+			return ret;
+
+		if (val >> 1)
+			return -EINVAL;
+
+		if (val) {
+			ret = lt8491_restart_chip(dev->dev);
+			if (ret)
+				return ret;
+		}
+
+		return len;
+	case LT8491_SCRATCH:
+		ret = iio_parse_value(buf, IIO_VAL_INT, &val, NULL);
+		if (ret)
+			return ret;
+
+		ret = lt8491_write_scratch(dev->dev, val);
+		if (ret)
+			return ret;
+
+		return len;
+	default:
+		return -EOPNOTSUPP;
+	}
+
+	if (ret)
+		return ret;
+
+	return len;
+}
+
+static struct iio_attribute lt8491_iio_attrs[] = {
+	LT8491_IIO_CH_ATTR_RO("tbat", LT8491_TBAT),
+	LT8491_IIO_CH_ATTR_RO("pout", LT8491_POUT),
+	LT8491_IIO_CH_ATTR_RO("pin", LT8491_PIN),
+	LT8491_IIO_CH_ATTR_RO("efficiency", LT8491_EFF),
+	LT8491_IIO_CH_ATTR_RO("iout", LT8491_IOUT),
+	LT8491_IIO_CH_ATTR_RO("iin", LT8491_IIN),
+	LT8491_IIO_CH_ATTR_RO("vbat", LT8491_VBAT),
+	LT8491_IIO_CH_ATTR_RO("vin", LT8491_VIN),
+	LT8491_IIO_CH_ATTR_RO("charging_stage", LT8491_CHARGING_STAGE),
+	LT8491_IIO_CH_ATTR_RO("charging_status", LT8491_CHARGING_STATUS),
+	LT8491_IIO_CH_ATTR_RW("enable", LT8491_ENABLE),
+	LT8491_IIO_CH_ATTR_WO("reset", LT8491_RESTART),
+	LT8491_IIO_CH_ATTR_RW("scratch", LT8491_SCRATCH),
+	LT8491_IIO_CH_ATTR_RO("serial_id", LT8491_SERIAL_ID),
+	END_ATTRIBUTES_ARRAY
+};
```

Comment:
These should all be debug attributes. "Standard" IIO attibutes are documented on the IIO ABI page https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-bus-iio and we should strive to use the same ones for no-OS too. Here is the list of attributes supported by no-os https://github.com/analogdevicesinc/no-OS/blob/main/iio/iio.c#L109 and a driver which uses both types https://github.com/analogdevicesinc/no-OS/blob/main/drivers/imu/iio_adis.c